File: .claude/api docs/on_provizyon_yeni/Kayitli Kart İşlemleri/NodeJS/preauth_kayitli_kart.js

// Mağaza bilgileri
var merchant_id = 'XXXXXX'
var merchant_key = 'XXXXXX'
var merchant_salt = 'XXXXXX'

// Kullanıcı bilgileri
var user_ip = '';
var email        = "testpreauth@example.com";
var user_name    = 'TEST TEST';
var user_address = 'TEST ADRES';
var   = user_phone = '5555555555';

var basket = JSON.stringify([
    ['Örnek Ürün 1', '18.00', 1],
    ['Örnek Ürün 2', '33.25', 2],
    ['Örnek Ürün 3', '45.42', 1] ]);// Ürün ismi, Fiyat, Adet



// İşlem bilgileri
var auth_type = 'preauth';
var merchant_oid      = 'TESTPREAUTH';
var currency          = 'TL';
var payment_amount    = '14.55';
var payment_type      = 'card';
var non_3d            = '1';
var non3d_test_failed = '0';
var installment_count = '1';// Ön provizyonda taksitli işlem yapılamaz. Bu nedenle 1 veya 0 gönderilmelidir.
var client_lang       = 'tr';
var test_mode         = '0'; //Test modunda işlemler şuan için desteklenmemektedir. test_mode "0" ileterek, işlemlerinizi canlı modda ve gerçek kart bilgileri ile gerçekleştiriniz.
var utoken         = '';
var ctoken         = '';


// paytr_token
var hashSTR = `${merchant_id}${user_ip}${merchant_oid}${email}${payment_amount}${payment_type}${installment_count}${currency}${test_mode}${non_3d}`;
    console.log('HASH STR' + hashSTR);
    var paytr_token = hashSTR + merchant_salt;
    console.log('PAYTR TOKEN' + paytr_token);
    var token = crypto.createHmac('sha256', merchant_key).update(paytr_token).digest('base64');
	
console.log('TOKEN' + token);
    context = {
		auth_type,
        merchant_id,
        user_ip,
        merchant_oid,
        email,
        payment_type,
        payment_amount,
        currency,
        test_mode,
        non_3d,
        user_name,
        user_address,
        user_basket,
		user_phone,
        client_lang,
        token,
        non3d_test_failed,
        installment_count,
		cc_owner,
		cc_number,
		cc_month,
		cc_year,
		cvv,
		utoken,
		ctoken,
		
    };

    res.render('index');
});
var callback = req.body;

    paytr_token = callback.merchant_oid + merchant_salt + callback.status + callback.total_amount;
    var token = crypto.createHmac('sha256', merchant_key).update(paytr_token).digest('base64');

  
var port = 3200;
app.listen(port, function () {
    console.log("Server is running. Port:" + port);
});

File: .claude/api docs/on_provizyon_yeni/Kayitli Kart İşlemleri/.NET/preauth_kayitli_kart.cs

// Mağaza bilgileri
string  = 'MAGAZA_NO';
string  = 'XXXXXXXXXXX';
string = 'YYYYYYYYYYY';

// Kullanıcı bilgileri
string ip = Request.ServerVariables["HTTP_X_FORWARDED_FOR"];
            if (user_ip == "" || user_ip == null){
                user_ip = Request.ServerVariables["REMOTE_ADDR"];
            }

string emailstr = "testpreauth@example.com";
string user_namestr    = 'TEST TEST';
string user_addressstr = 'TEST ADRES';
string user_phonestr   = '5555555555';
string user_ip  = $ip;
object[][] user_basket = {
            new object[] {"Örnek ürün 1", "18.00", 1}, // 1. ürün (Ürün Ad - Birim Fiyat - Adet)
            new object[] {"Örnek ürün 2", "33.25", 2}, // 2. ürün (Ürün Ad - Birim Fiyat - Adet)
            new object[] {"Örnek ürün 3", "45.42", 1}, // 3. ürün (Ürün Ad - Birim Fiyat - Adet)
            }; // Ürün ismi, Fiyat, Adet




// İşlem bilgileri
string merchant_oid      = 'TESTPREAUTH';
string currency          = 'TL';
string payment_amount    = '14.55';
string payment_type      = 'card';
string non_3d            = '1';
string non3d_test_failed = '0';
string installment_count = '1';// Ön provizyonda taksitli işlem yapılamaz. Bu nedenle 1 veya 0 gönderilmelidir.
string client_lang       = 'tr';
string test_mode         = '0'; //Test modunda işlemler şuan için desteklenmemektedir. test_mode "0" ileterek, işlemlerinizi canlı modda ve gerçek kart bilgileri ile gerçekleştiriniz.
string utoken         = '';
string ctoken         = '';

 JavaScriptSerializer ser = new JavaScriptSerializer();
  string user_basket_json = ser.Serialize(user_basket);

// paytr_token
string Birlestir = string.Concat(merchant_id, user_ip, merchant_oid, emailstr, payment_amountstr.ToString(), payment_type, installment_count, currency, test_mode, non_3d, merchant_salt);
HMACSHA256 hmac = new HMACSHA256(Encoding.UTF8.GetBytes(merchant_key));
byte[] b = hmac.ComputeHash(Encoding.UTF8.GetBytes(Birlestir));

			ViewBag.auth_type = 'preauth';
			ViewBag.MerchantId = merchant_id;
            ViewBag.UserIp = user_ip;
            ViewBag.MerchantOid = merchant_oid;
            ViewBag.Email = emailstr;
            ViewBag.PaymentType = payment_type;
            ViewBag.PaymentAmount = payment_amountstr.ToString();
            ViewBag.InstallmentCount = installment_count;
            ViewBag.Currency = currency;
            ViewBag.TestMode = test_mode;
            ViewBag.Non3d = non_3d;
			ViewBag.Client_Lang = client_lang;
            ViewBag.UserName = user_namestr;
            ViewBag.UserAddress = user_addressstr;
            ViewBag.UserPhone = user_phonestr;
            ViewBag.UserBasket = user_basket_json;
            ViewBag.Non3dTestFailed = non3d_test_failed;
            ViewBag.PaytrToken = Convert.ToBase64String(b);
			ViewBag.cc_owner = cc_owner;
			ViewBag.cc_number= cc_number;
			ViewBag.cc_month = cc_month;
			ViewBag.cc_year= cc_year;
			ViewBag.cvv = cvv;
			ViewBag.utoken = utoken;
			ViewBag.ctoken = ctoken;

            return View();
        }
    }
}


